test_runner/fixtures/utils.py

```diff
@@ -37,6 +37,58 @@
 
 
 Fn = TypeVar("Fn", bound=Callable[..., Any])
+COMPONENT_BINARIES = {
+    "storage_controller": ["storage_controller"],
+    "storage_broker": ["storage_broker"],
+    "compute": ["compute_ctl"],
+    "safekeeper": ["safekeeper"],
+    "pageserver": ["pageserver", "pagectl"],
+}
+
+VERSIONS_COMBINATIONS = [
+    {
+        "storage_controller": "new",
+        "storage_broker": "new",
+        "compute": "new",
+        "safekeeper": "new",
+        "pageserver": "new",
+    },
+    {
+        "storage_controller": "old",
+        "storage_broker": "old",
+        "compute": "old",
+        "safekeeper": "old",
+        "pageserver": "new",
+    },
+    {
+        "storage_controller": "old",
+        "storage_broker": "new",
+        "compute": "old",
+        "safekeeper": "new",
+        "pageserver": "old",
+    },
+    {
+        "storage_controller": "new",
+        "storage_broker": "old",
+        "compute": "new",
+        "safekeeper": "old",
+        "pageserver": "old",
+    },
+    {
+        "storage_controller": "new",
+        "storage_broker": "old",
+        "compute": "old",
+        "safekeeper": "new",
+        "pageserver": "old",
+    },
+    {
+        "storage_controller": "old",
+        "storage_broker": "new",
+        "compute": "new",
+        "safekeeper": "old",
+        "pageserver": "old",
+    },
+]
```

Comment:
The previous version had better readability, let's bring it back.
We can let auto-formatter to accept if this is the problem like this: https://github.com/neondatabase/neon/blob/c2623ffef454378b2602f494e459b32028aa04a0/test_runner/performance/test_perf_olap.py#L107-L125

Reply:
I got it, thanks